### ISSUE TITLE
Synchronizer thread does not work with preforked servers

### ISSUE THREAD

**Issue by feljen** [splitio/python-client#177]
Hi, we are using the in-memory split client for our flask apps. When running uwsgi in master mode we noticed that splits do not receive updates made in the UI. I believe this is because the polling thread is started in the master process and does not have access to the worker process's memory space.

We are able to temporarily get around this issue by using uwsgi's `--lazy-app` flag, but there is a concern around increased memory consumption and we don't want to do this long-term.

I think the simplest solution would be to make sure the synchronizer thread is started after the worker process is forked. Other python preforked servers like gunicorn will run into this issue as well. We also want to eliminate the need to go over network to talk to a Redis instance (reduce the chance of network failures/latency), which is why we want to go with the in-memory option.

**Comment by chillaq** [comment]
Hi Jenna, With the uwsgi server I am not sure if it is possible to use SDK in-memory cache, due to the issue you mention.
You can use the Redis cache and Synchronizer app with enabling "redisLocalCacheEnabled" SDK parameter, this will reduce the amount of network traffic between Python and redis if the same Split is being used again in getTreatment call.

Let me know if you have tried this option before.

Thanks
Bilal

**Comment by feljen** [comment]
Thanks for your reply. We've explored the redis option previously but I'm actually hoping this can be a small change to the SDK to have each worker spawn its own thread, i.e. doing something like:
```
AsyncTask(object):

def __init__(self):
   # set pid in the constructor
   self._pid = None

def start(self):
   current_pid = os.getpid()
   # only start thread if detect a change in pid
   if self._pid != current_pid:
      thread.start()
      self._pid = current_pid
```
Would something like that work?

**Comment by chillaq** [comment]
Hi Jenna, we will do some testing and validation for this approach and let you know how it goes.

Thanks
Bilal

**Comment by feljen** [comment]
Any updates on this?

**Comment by chillaq** [comment]
Hi Jenna, no updates yet, will ping engineering.

**Comment by chillaq** [comment]
Hi Jenna, we implemented the changes in Python SDK version 8.4.0

Thanks
Bilal
